servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletRegistration.java

```diff
@@ -0,0 +1,252 @@
+import javax.servlet.ServletSecurityElement;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.server.servlet.util.UrlMapper;
+
+/**
+ * The servlet supportPipeline.
+ */
+public class ServletRegistration implements Dynamic {
+    private final String servletName;
+    private final Servlet servlet;
+    private final ServletConfig servletConfig;
+    private final DefaultServletContext servletContext;
+    private final UrlMapper<ServletRegistration> urlMapper;
+    private final Set<String> mappingSet = new HashSet<>();
+    private final Map<String, String> initParameterMap;
+
+    private boolean asyncSupported = true;
+    private int loadOnStartup = -1;
+    private AtomicBoolean initServlet = new AtomicBoolean();
+
+    @Nullable
+    private MultipartConfigElement multipartConfigElement;
+    @Nullable
+    private ServletSecurityElement servletSecurityElement;
+    @Nullable
+    private String roleName;
+
+    /**
+     * Creates a new instance.
+     */
+    public ServletRegistration(String servletName, Servlet servlet, DefaultServletContext servletContext,
+                               UrlMapper<ServletRegistration> urlMapper) {
+        this(servletName, servlet, servletContext, urlMapper, ImmutableMap.copyOf(new HashMap<>()));
+    }
+
+    /**
+     * Creates a new instance.
+     */
+    public ServletRegistration(String servletName, Servlet servlet, DefaultServletContext servletContext,
+                               UrlMapper<ServletRegistration> urlMapper, Map<String, String> initParameterMap) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(servlet, "servlet");
+        requireNonNull(servletContext, "servletContext");
+        requireNonNull(urlMapper, "urlMapper");
+
+        this.servletName = servletName;
+        this.servlet = servlet;
+        this.servletContext = servletContext;
+        this.urlMapper = urlMapper;
+        this.initParameterMap = ImmutableMap.copyOf(initParameterMap);
+        servletConfig = new ServletConfig() {
+            @Override
+            public String getServletName() {
+                return ServletRegistration.this.servletName;
+            }
+
+            @Override
+            public ServletContext getServletContext() {
+                return ServletRegistration.this.servletContext;
+            }
+
+            @Override
+            @Nullable
+            public String getInitParameter(String name) {
+                return ServletRegistration.this.getInitParameter(name);
+            }
+
+            @Override
+            public Enumeration<String> getInitParameterNames() {
+                return Collections.enumeration(ImmutableSet.copyOf(getInitParameters().keySet()));
+            }
+        };
+    }
+
+    /**
+     * Get servlet security element.
+     */
+    @Nullable
+    public ServletSecurityElement getServletSecurityElement() {
```

Comment:
When this method is used? If this is not used now, can we add this when we need?